src/service/submit.rs

```diff
@@ -28,46 +28,55 @@ pub fn update_submit_parameters(
 }
 
 pub fn submit(res: &mut Resources<impl RP>, filename: String) -> Result<u64> {
-    require_login(res)?;
-    let storage = res.storage.get();
-    let course_id = storage
-        .get_course()
-        .ok_or_else(|| anyhow!("Course not provided"))?
-        .to_owned();
-    let task_id = storage
-        .get_task()
-        .ok_or_else(|| anyhow!("Task not provided"))?
-        .to_owned();
-    let language_name = storage
-        .get_language()
-        .ok_or_else(|| anyhow!("Language not provided"))?
-        .to_owned();
-    let language_option = storage.get_option().map(|t| t.to_owned());
+    (|| -> Result<u64> {
```

Comment:
Specifying return type shouldn't be necessary
```suggestion
    (|| {
```

Reply:
Leaving the return type out results in compilation error as operator `?` needs to know how to cast the error.